tensorflow_addons/layers/noisy_dense.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import tensorflow as tf
+from tensorflow.keras import (
+    activations,
+    initializers,
+    regularizers,
+    constraints,
+)
+from tensorflow.keras import backend as K
+from tensorflow.keras.layers import InputSpec
+from typeguard import typechecked
+
+from tensorflow_addons.utils import types
+
+
+def _scale_noise(x):
+    return tf.sign(x) * tf.sqrt(tf.abs(x))
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class NoisyDense(tf.keras.layers.Dense):
+    r"""Noisy dense layer that inject random noise to
+    the weights of normal dense layer.
```

Comment:
Is this formatted by our formatter? It should be one line so the doc rendering works correctly. Maybe we can use less words like
```
r"""Noisy dense layer.

Describe some details of the this layers in plain text. Maybe use the section 3 in the paper.
See [Noisy Networks for Explanation](https://arxiv.org/pdf/1706.10295.pdf) for more details.
```
And put the whole descriptions below it.

Reply:
Flake8 was complaining that the line was too long so I split it into 2. 

I could make it:
```
Noisy dense layer that injects random noise to the weights of dense layer.
```

instead of:
```
Noisy dense layer that injects random noise to the weights of normal dense layer.
```

Reply:
Sure. LGTM :-) Don't forget to update the detailed descriptions and paper link down below.